app/src/containers/Signup/Signup.js

```diff
@@ -1,9 +1,154 @@
+    let confirmationValid = this.state.confirmationValid;
+
+    switch (fieldName) {
+      case 'email':
+        emailValid = value.match(mail);
+        isErrors.email = emailValid ? '' : `${messages.emailError}`;
+        break;
+      case 'password':
+        passwordValid = value.match(pass);
+        isErrors.password = passwordValid ? '' : `${messages.passwordError}`;
+        break;
+      case 'confirmation':
+        isErrors.confirmation = 
+          (this.state.confirmation === this.state.password) ? '' : `${messages.confirmationError}`;
+        break;
+      default:
+        break;
+    }
+
+    this.setState({isErrors: isErrors,
+                    emailValid: emailValid,
+                    passwordValid: passwordValid,
+                  }, this.isValidate);
+  }
+
+  isValidate = () => {
+    this.setState({ 
+      Validate:
+        this.state.emailValid && 
+        this.state.passwordValid && 
+        this.state.confirmation === this.state.password
+    });
+  }
+
+  errorClass = (error) => {
+    return (error.length === 0 ? '' : 'has-error');
+  }
+
+  handleButtonClick = e => {
+    e.preventDefault();
+    this.props.actions.signupUser(this.state.email, this.state.password)
+  }
+
+  render() {
     return (
-      <h1>Here will be signup page.</h1>
+      <div className='containerLog'>
+      <Message />
+        <Form className='Signup' onSubmit={this.handleButtonClick}>
+          {['first_Name', 'last_Name'].map(param =>
+            <FieldGroup
+              key={param}
+              label={param[0].toUpperCase() + param.slice(1).replace('_', ' ')}
+              type={'text'}
+              name={param}
+              placeholder={'Enter ' + param[0].toUpperCase() +
+              param.slice(1).replace('_', ' ')}
+              value={this.state.param}
+              onChange={this.handleChange(param)}
+            />
+          )}
+          {['email', 'password', 'confirmation'].map(param =>
+            <FieldGroup
+              key={param}
+              className={`${this.errorClass(this.state.isErrors[param])}`}
+              label={param[0].toUpperCase() +
+                param.slice(1).replace('onfirmation', 'onfirmation Password') +
+                '*'}
+              type={param.replace('confirmation', 'password')}
+              name={param}
+              isErrors={this.state.isErrors[param]}
+              placeholder={'Enter ' + param[0].toUpperCase() +
+                param.slice(1).replace('onfirmation', 'onfirmation Password')}
+              value={this.state.param}
+              onChange={this.handleChange(param)}
+              required
+            />
+          )}
+          <Button 
+            className='btn btn-primary'
+            disabled={!this.state.Validate}
+            type='submit'>Regist
+          </Button>
+        </Form>
+      </div>
     );
   }
 }
+
+const mapStatetoProps = (state, ownProps) => {
+  return {
```

Comment:
With arrow functions you can return object without word "return": => ({val: prop})